File: logger.java
// TC :O(1)
// SC :O(n)

class Logger {
    HashMap<String,Integer> map;
    public Logger() {
        map = new HashMap<>();
    }
    
    public boolean shouldPrintMessage(int timestamp, String message) {
        if(!map.containsKey(message)){
            map.put(message,timestamp);
            return true;
        }
        else{
            int time = map.get(message);
            if(time+10<=timestamp){
                map.put(message,timestamp);
                return true;
            }
        }
        return false;
    }
}

/**
 * Your Logger object will be instantiated and called as such:
 * Logger obj = new Logger();
 * boolean param_1 = obj.shouldPrintMessage(timestamp,message);
 */

File: logger.cpp
//TC: O(1)
//SC: O(M) where M is all messages received

class Logger {
public:
    /** Initialize your data structure here. */
    
    unordered_map<string, int> logmap;
    
    Logger() {
    
    }
    
    /** Returns true if the message should be printed in the given timestamp, otherwise returns false.
        If this method returns false, the message will not be printed.
        The timestamp is in seconds granularity. */
    bool shouldPrintMessage(int timestamp, string message) {
        
        if(logmap.find(message)!=logmap.end()){
            int currentTime = logmap[message];
            if(timestamp - currentTime >=10){
                logmap[message] = timestamp;
                return true;
            } else{
                return false;
            }
            
        } else{
            logmap[message] = timestamp;
            return true;
        }
        
        
    }
};

/**
 * Your Logger object will be instantiated and called as such:
 * Logger* obj = new Logger();
 * bool param_1 = obj->shouldPrintMessage(timestamp,message);
 */